src/prompt/interaction.rs

```diff
@@ -99,8 +99,7 @@ pub trait PromptInteraction<T> {
             }
 
             match term.read_key() {
-                Ok(Key::Escape) => state = State::Cancel,
-
+                Ok(Key::Escape) if self.input().is_none() => state = State::Cancel,
```

Comment:
Better don't touch the common code, you just added a few bugs here (which we don't know yet). It's about Input multiline: thus, modify it in the Input module - if it's in edit mode, then handle `Esc`, and quit. It should work...

Reply:
```
Ok(Key::Escape) => state = State::Cancel,
Ok(key) => {
     ... 
    state = self.on(&Event::Key(key));
}
```

If key is `Escape`, it will end in the first branch. If I want to handle it, maybe I have to remote the first branch. Then others can still not work...

Reply:
Yes, you are right, I know how to fix this now.

Reply:
f587db5

Maybe the solution is not good, but it works at least.

Reply:
@kingwingfly I see the problem, it actually is a problem with the original `on() -> State` API. It lacks the "pass the control over" functionality. If it had been changed to `on() -> Option<State>` (`Some` for overriding the state, and `None` for passing the control over to the outer loop) then it would've worked for us: `Some(State::Cancel)` would mean an actual canceling, and `None` - "I don't care, let's the outer loop decide".

Alright, now, a little problem is that your patch is based on the refactoring I haven't accepted. I played with the cancellation and I've come up with a similar workaround: https://github.com/fadeevab/cliclack/pull/59

Reply:
@kingwingfly Thanks for pointing this out!

I analyzed your patch, and the corner cases are basically one-line fixes:

1. View mode with default value: it's an assignment of the multiline mode in the `interaction()`.
2. The key hit miss is fixed with `self.input().insert(c)`.

I used your patch to fix my patch: https://github.com/fadeevab/cliclack/pull/59/files

BTW, I found that multiline editing is either disabled, "enabled in editing mode", or "enabled in preview mode". So I turned it into an enum, check it out.